### DISCUSSION TITLE
I can't show an error, what should I do?

### DISCUSSION THREAD

**Discussion by goodwin74** [cefsharp/CefSharp#3326] [General]
**I am using: WinForms/ Lang C# / CefSharp 86 / VS2019**

I am trying to show an error when the request fails, according to the examples.
```
browser.LoadError += (obj, args) =>
            {
                var errorBody = "<html><body><h2 style='width:100%;text-align:center;padding-top:100px;'>Error</h2></body></html>";
                
                browser.LoadHtml(errorBody,args.FailedUrl,Encoding.UTF8,true);
            };
```

But this code doesn't work. On error, the browser goes into an infinite loop. It seems to me that LoadHtml itself raises an error again and circle is closed.
What to do and how to show the error correctly?
I've also tried using LoadHtml (html, base64) and it works, but the problem is the url gets ugly.
Is there any solution to load the error from FailedUrl?

**Comment by amaitland** [comment]
Firstly I'd suggest ignoring `Aborted` requests

```c#
// Don't display an error for downloaded files.
if (args.ErrorCode == CefErrorCode.Aborted)
{
    return;
}
```



> On error, the browser goes into an infinite loop. It seems to me that LoadHtml itself raises an error again and circle is closed.

Have you added a break point and checked what error code is being sent the second time? Sounds like your speculating rather than you've actually debugged and know for sure what's going on.

> I've also tried using LoadHtml (html, base64) and it works, but the problem is the url gets ugly.

This is the approach I recommend and use. Just don't update your `Address` bar (or whaterver you use to display the url to your user) when the address is a `Data Uri`.

See https://github.com/cefsharp/CefSharp/issues/2985 for a discussion of the limitations of using `LoadHtml` to display errors. Until `CEF` supports `Interstitial pages` there's not a whole lot we can do.

**Reply by goodwin74** [reply]
It is a shame that CEF does not support this feature(show special pages in case of errors). Do you know if they have plans to create such an opportunity?

**Reply by amaitland** [reply]
CEF has a feature request forum at https://magpcss.org/ceforum/viewforum.php?f=7

You can pose adding support and ask if the CEF maintainers are planning to add support.
